src/legacy/core_plugins/expressions/public/np_ready/public/render.ts

```diff
@@ -33,59 +33,91 @@ export class ExpressionRenderHandler {
   private element: HTMLElement;
   private destroyFn?: any;
   private renderCount: number = 0;
+  private renderSubject: Rx.Subject<unknown>;
+  private eventsSubject: Rx.Subject<unknown>;
+  private updateSubject: Rx.Subject<unknown>;
   private handlers: IInterpreterRenderHandlers;
 
   constructor(element: HTMLElement) {
     this.element = element;
 
-    const eventsSubject = new Rx.Subject();
-    this.events$ = eventsSubject.asObservable().pipe(share());
+    this.eventsSubject = new Rx.Subject();
+    this.events$ = this.eventsSubject.asObservable().pipe(share());
 
-    const renderSubject = new Rx.Subject();
-    this.render$ = renderSubject.asObservable().pipe(share());
+    this.renderSubject = new Rx.Subject();
+    this.render$ = this.renderSubject.asObservable().pipe(share());
 
-    const updateSubject = new Rx.Subject();
-    this.update$ = updateSubject.asObservable().pipe(share());
+    this.updateSubject = new Rx.Subject();
+    this.update$ = this.updateSubject.asObservable().pipe(share());
 
     this.handlers = {
       onDestroy: (fn: any) => {
         this.destroyFn = fn;
       },
       done: () => {
         this.renderCount++;
-        renderSubject.next(this.renderCount);
+        this.renderSubject.next(this.renderCount);
       },
       reload: () => {
-        updateSubject.next(null);
+        this.updateSubject.next(null);
       },
       update: params => {
-        updateSubject.next(params);
+        this.updateSubject.next(params);
       },
       event: data => {
-        eventsSubject.next(data);
+        this.eventsSubject.next(data);
       },
     };
   }
 
-  render = async (data: Data, extraHandlers: IExpressionRendererExtraHandlers = {}) => {
-    if (!data || data.type !== 'render' || !data.as) {
-      throw new Error('invalid data provided to expression renderer');
+  render = (data: Data, extraHandlers: IExpressionRendererExtraHandlers = {}) => {
+    if (!data || typeof data !== 'object') {
+      this.renderSubject.next({
```

Comment:
sorry, i probably wasn't explaining well what i meant here, i think its ok if we `this.renderSubject.error()` here, i just wanted the data$ to not error (so we don't have to listen to two observables to figure out what happened)

Reply:
This is definitely not the right behavior, because as soon as we call `renderSubject.erorr()` it stops the observable.